### ISSUE TITLE
Interface in Java and its implementation

### ISSUE THREAD

**Issue by gracy20** [section-engineering-education/engineering-education#3507]
### Proposed title of article
Interface in Java and its implementation

### Introduction paragraph (2-3 paragraphs):

In the Java programming language, an interface is an abstract type that specifies a class's behavior. The interface keyword is used to declare interfaces in Java. In Java interface is a blueprint of a class. It features abstract methods and static constants.
In Java, the interface is a means of achieving abstraction. The Java interface can only have abstract methods, no method bodies. In Java, it is used to achieve abstraction as well as multiple inheritance.
To put it another way, interfaces can have abstract methods and variables. It isn't allowed to have a method body.

### Key takeaways: 
1. Reasons to use interface
2. Declaring an interface
3. Abstract class vs Interface
4. Multiple inheritances by interface and why it is not supported through class in java
5. Interface inheritance
6. Default and Static Method in Interface

### Reference:
N/A
